rebuild/src/public_key.cc

```diff
@@ -0,0 +1,154 @@
+#include "public_key.h"
+
+void PublicKey::Init(
+    Napi::Env env, Napi::Object &exports, BlstTsAddon *module) {
+    Napi::HandleScope scope(env);
+    auto proto = {
```

Comment:
In general, I like using `auto` everywhere I can. Here, I'm a bit puzzled - you used it for the `proto`, but not for other variables. It might make sense to be consistent here, whichever _style_ it may be.

Reply:
> In general, I like using `auto` everywhere I can. Here, I'm a bit puzzled - you used it for the `proto`, but not for other variables. It might make sense to be consistent here, whichever _style_ it may be.

I actually tend to go the other way and prefer hard typing everywhere.  This type just happens to be a brutal one and changes with each method that gets added so i went with auto in this one locations.


> Also, having this `proto` static might make sense, given it's initialized the same way for every invocation. This can avoid extra CPU cycles.

Statics are big no-no's in Node.  In particular anything with a `Napi` or any data that is non-const.  Node is technically a multithreaded environment in certain situations (Electron Apps, Worker Threads, Child Processes).  So there are lots of instances where there are multiple versions of the library loaded.

The sticky wicket is that node loads all native modules as dylibs and only once per process, but initiialized them for each thread separately (to maintain the single threaded nature of node).  So there are race conditions that happen when attempting to use data across threads in an unsafe way.  Or worse using Napi (node created/managed) objects in an instance other than the one that manages the instance.  Often ends up in segfault (but not always) do to the nature of how the objects are referenced.

Reply:
3a5f64fb9669714ec8c4b5ecc3d511f0042ad8a3